components/ws-daemon/pkg/content/service.go

```diff
@@ -213,27 +212,19 @@ func (s *WorkspaceService) InitWorkspace(ctx context.Context, req *api.InitWorks
 			UID:     wsinit.GitpodUID,
 			GID:     wsinit.GitpodGID,
 		}
-		err = RunInitializer(ctx, workspace.Location, req.Initializer, remoteContent, opts)
-		if err != nil {
-			log.WithError(err).WithField("workspaceId", req.Id).Error("cannot initialize workspace")
-			return nil, status.Error(codes.Internal, fmt.Sprintf("cannot initialize workspace: %s", err.Error()))
-		}
-
 		if req.UserNamespaced {
 			// This is a bit of a hack as it makes hard assumptions about the nature of the UID mapping.
 			// Also, we cannot do this in wsinit because we're dropping all the privileges that would be
 			// required for this operation.
 			//
 			// With FWB this bit becomes unneccesary.
-			var (
-				uid = wsinit.GitpodUID + 100000 - 1
-				gid = wsinit.GitpodGID + 100000 - 1
-			)
-			out, err := exec.Command("chown", "-R", fmt.Sprintf("%d:%d", uid, gid), workspace.Location).CombinedOutput()
-			if err != nil {
-				log.WithError(err).WithField("workspaceId", req.Id).WithField("msg", string(out)).Error("cannot chown workspace")
-				return nil, status.Error(codes.Internal, "cannot prepare workspace")
-			}
+			opts.UID = wsinit.GitpodUID + 100000 - 1
+			opts.GID = wsinit.GitpodGID + 100000 - 1
+		}
+		err = RunInitializer(ctx, workspace.Location, req.Initializer, remoteContent, opts)
```

Comment:
Out of curiosity, what does the "non-shifted backup" means in that context?

> Because we still give the content-inititalizer CAP_FCHOWN that also works with the non-shifted backups we have. 

This change looks nice so far. But I wasn't digging into the internals of the initializer to see, what's happening behind the scene with the backups, esp. to understand the difference between them and a "regular" content.

Reply:
When we create the backup we undo the UID shift, i.e. in there files will belong to `33333`.
The initializer runs with UID `13332` and needs chown files correctly, hence CAP_FCHOWN